ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -699,3 +703,19 @@ def setup_ceph_debug():
     )
     log.info("Setting Ceph to work in debug log level using a new ConfigMap resource")
     run_cmd(f"oc create -f {ceph_configmap_yaml.name}")
+
+
+def verify_sc_images(storage_cluster):
+    """
+    Verifying images in storage cluster such as ceph, noobaaDB and noobaaCore
+
+    Args:
+        storage_cluster (obj): storage_cluster ocp object
+    """
+    images_set = set()
+    images = storage_cluster.get().get("status").get("images")
+    for component, images_dict in images.items():
+        for image, image_name in images_dict.items():
+            log.info(f"{component} has {image}:{image_name}")
+            images_set.add(image_name)
+    assert len(images_set) == 3, "actualImage and desiredImage are different"
```

Comment:
What if noobaa is disabled - https://github.com/red-hat-storage/ocs-ci/pull/4148/files when flexible component feature is in in 4.8 ? Please verify on 4.8 with disabled noobaa - it should work even in 4.7 from what I remember from @shylesh

Reply:
@petr-balogh you are right , if noobaa is disabled output could be different. noobaa operator will be there but core and db pods will not be there but I am not sure what would be the output from storage cluster. 

@shreekarSS  could you please confirm this. Just run deployment with --disable-components noobaa and check image versions.

Reply:
Thanks @petr-balogh @shylesh , Tested the script with --disable-components noobaa  and updated the logic to handle disabled components images as well. 
With --disable-components noobaa :
images:
    ceph:
      actualImage: quay.io/rhceph-dev/rhceph@sha256:725f93133acc0fb1ca845bd12e77f20d8629cad0e22d46457b2736578698eb6c
      desiredImage: quay.io/rhceph-dev/rhceph@sha256:725f93133acc0fb1ca845bd12e77f20d8629cad0e22d46457b2736578698eb6c
    noobaaCore:
      desiredImage: quay.io/rhceph-dev/mcg-core@sha256:7b7292c8e62c3a86228745852f5c696fa505ba3c6f6f06fe0de61f8fb2ca2589
    noobaaDB:
      desiredImage: registry.redhat.io/rhel8/postgresql-12@sha256:03a1e02a1b3245f9aa0ddd3f7507b915a8f7387a1674969f6ef039a5d7fd8bf0

Will be ignoring components without "actualImage", taking components with actual and desired images and checking if they match.